### PR TITLE
VTK: add -no-ipo for builds using intel compiler

### LINKED ISSUE #20471: Installation issue: VTK, internal compiler error with Intel 19.0.5
### Steps to reproduce the issue

```console
$ spack install vtk@8.2.0 ~mpi +opengl2 %intel ^freetype%gcc ^hdf5~mpi
...
  >> 1846    ": internal error: ** The compiler has encountered an unexpected problem.
     1847    ** Segmentation violation signal raised. **
     1848    Access violation or stack overflow. Please contact Intel Support for assistance.
     1849
  >> 1850    icpc: error #10014: problem during multi-file optimization compilation (code 4)
  >> 1851    make[2]: *** [lib/libvtkCommonMath-8.2.so.1] Error 4
     1852    make[2]: Leaving directory `/tmp/bamboo_sta/spack-stage/spack-stage-vtk-8.2.0-jugxugntvxvvhv2hm
             qadjumisr4xs7sn/spack-build-jugxugn'
  >> 1853    make[1]: *** [Common/Math/CMakeFiles/vtkCommonMath.dir/all] Error 2
```

### Information on your system

* **Spack:** 0.16.0
* **Python:** 3.6.8
* **Platform:** linux-centos7-broadwell
[spack-build-out.txt](https://github.com/spack/spack/files/5714900/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/5714901/spack-build-env.txt)

### Additional information

* [spack-build-out.txt]()
* [spack-build-env.txt]()

@chuckatkins @danlipsa 

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by andymwood**
Fixes #20471, avoiding an internal compiler error.

### CODE REVIEW

File: var/spack/repos/builtin/packages/vtk/package.py [hunk 1]
```diff
@@ -298,4 +297,16 @@ def cmake_args(self):
...
+        # -no-ipo prevents an internal compiler error from multi-file
+        # optimization (https://github.com/spack/spack/issues/20471)
+        if '%intel' in spec:
+            compile_flags.append('-no-ipo')
+
+        if compile_flags:
+            compile_flags = ' '.join(compile_flags)
+            cmake_args.extend([
+                '-DCMAKE_C_FLAGS="{}"'.format(compile_flags),
+                '-DCMAKE_CXX_FLAGS="{}"'.format(compile_flags)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/vtk/package.py]:
```suggestion
                '-DCMAKE_C_FLAGS={0}'.format(compile_flags),
                '-DCMAKE_CXX_FLAGS={0}'.format(compile_flags)
```
Needed for Python 2.6 support

**Review by adamjstewart — CHANGES_REQUESTED** [review]
@chuckatkins @danlipsa can you review?

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Double quotes shouldn't be needed. If you were building this on the command line, bash would strip the double quotes, but we aren't using bash.

**Comment by tldahlgren** [comment]
Thank you for your contribution!

**Pull request merged**